.semaphore/semaphore.yml

```diff
@@ -19,60 +19,164 @@ queue:
-          . vault-setup
-        fi
-  epilogue:
-    always:
-      commands:
-        - make ci-bin-sem-cache-store
-        - make store-test-results-to-semaphore
-
 blocks:
-  - name: "Build & Test (VS Code)"
+  # --- Build & Test (multi-platform/-arch) for VS Code (stable) ---
+  - name: "Build & Test (VS Code: linux x64)"
     dependencies: []
     # Skip when:
     # - change in the release.svg file, signaling a `chore: none version bump vN.N.N` commit.
     # - change in the .versions/next.txt file, signaling a release PR merge commit.
-    skip:
+    skip: &build-test-skip-stable
       when: "change_in(['/release.svg', '/.versions/next.txt'], {pipeline_file: 'ignore', branch_range: '$SEMAPHORE_GIT_COMMIT_RANGE', default_branch: 'main'})"
     task:
-      jobs:
+      agent:
+        machine:
+          type: s1-prod-ubuntu24-04-amd64-1
+      prologue: &build-test-prologue
+        commands:
+          - checkout
+          - make ci-bin-sem-cache-restore
+          - |
+            if [[ "${SEMAPHORE_ORGANIZATION_URL}" == *".semaphoreci.com" ]]; then
+              echo "Skipping Vault setup for Semaphore CI"
+            else
+              . vault-setup
+            fi
+      jobs: &build-test-jobs-stable
         - name: "Build & Test (VS Code)"
           commands:
             - make test
-          # Not setting VSCODE_VERSION env var will default using the latest stable release version
-      epilogue:
+          env_vars:
+            - name: VSCODE_VERSION
+              value: stable
+      epilogue: &build-test-epilogue
         always:
           commands:
             - make remove-test-env
-  - name: "Build & Test (VS Code Insiders)"
+            - make ci-bin-sem-cache-store
+            - make store-test-results-to-semaphore
+
+  - name: "Build & Test (VS Code: linux arm64)"
     dependencies: []
-    skip:
-      when: "branch != 'main' AND change_in(['/release.svg', '/.versions/next.txt'], {pipeline_file: 'ignore', branch_range: '$SEMAPHORE_GIT_COMMIT_RANGE', default_branch: 'main'})"
+    skip: *build-test-skip-stable
+    task:
+      agent:
+        machine:
+          type: s1-prod-ubuntu24-04-arm64-1
+      prologue: *build-test-prologue
+      jobs: *build-test-jobs-stable
+      epilogue: *build-test-epilogue
+
+  # Disabled until we can connect to an active GUI session
+  # - name: "Build & Test (VS Code: darwin x64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-amd64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
+
+  # - name: "Build & Test (VS Code: darwin arm64)"
+  #   dependencies: []
+  #   skip: *build-test-skip-stable
+  #   task:
+  #     agent:
+  #       machine:
+  #         type: s1-prod-macos-13-5-arm64
+  #     prologue: *build-test-prologue
+  #     jobs: *build-test-jobs-stable
+  #     epilogue: *build-test-epilogue
```

Comment:
I'd be in favor of removing this and linking a GH issue that contains this YAML.

Reply:
Sounds good, added to https://github.com/confluentinc/vscode/issues/1247